packages/node/src/indexer/api.service.ts

```diff
@@ -85,81 +85,100 @@ export class ApiService
       );
     }
 
-    const endpointToApiIndex: Record<string, ApiPromiseConnection> = {};
-
     if (chainTypes) {
       logger.info('Using provided chain types');
     }
 
     for await (const [i, endpoint] of network.endpoint.entries()) {
-      const connection = await ApiPromiseConnection.create(
-        endpoint,
-        this.fetchBlocksBatches,
-        {
-          chainTypes,
-        },
+      const connection = await this.createAndValidateConnection(endpoint, i);
```

Comment:
Is there any way we can move some of this to node core? Ideally we don't need to make changes to node so that we can easily update other SDKs